cmd/podman/pod_kill.go

```diff
@@ -64,31 +56,7 @@ func podKillCmd(c *cli.Context) error {
 		killSignal = uint(sysSignal)
 	}
 
-	if c.Bool("all") {
-		pods, err = runtime.Pods()
-		if err != nil {
-			return errors.Wrapf(err, "unable to get pods")
-		}
-	}
-	if c.Bool("latest") {
-		pod, err := runtime.GetLatestPod()
-		if err != nil {
-			return errors.Wrapf(err, "unable to get latest pod")
-		}
-		pods = append(pods, pod)
-	}
-	for _, i := range args {
-		pod, err := runtime.LookupPod(i)
-		if err != nil {
-			logrus.Errorf("%q", lastError)
-			if lastError != nil {
-				logrus.Errorf("%q", lastError)
-			}
-			lastError = errors.Wrapf(err, "unable to find pods %s", i)
-			continue
-		}
-		pods = append(pods, pod)
-	}
+	pods, lastError := checkFlagsAndGetPods(c, runtime)
```

Comment:
Should we return lastError here right away of len(pods) == 0?

Reply:
I don't think so. The loop will just be skipped and it will fall through, so I feel like it'd be a needless check.

Reply:
I think we might want to keep checkMutuallyExclusiveFlags and getPods separate - so we can make sure checkMutuallyExclusiveFlags runs before we create the runtime, and we give back feedback about invalid command line arguments early

Reply:
The reason I would like to check the length of pods is for the reason that @mheon and @TomSweeneyRedHat are pointing out, as well as the problem that someone at some future point, might assume that stuff was successful.